user-interface/src/main/java/life/qbic/datamanager/files/parsing/converters/ConverterFactory.java

```diff
@@ -0,0 +1,75 @@
+package life.qbic.datamanager.files.parsing.converters;
+
+import java.util.Map;
+import java.util.function.Supplier;
+import life.qbic.projectmanagement.application.api.AsyncProjectService.MeasurementRegistrationInformationNGS;
+import life.qbic.projectmanagement.application.api.AsyncProjectService.MeasurementRegistrationInformationPxP;
+import life.qbic.projectmanagement.application.api.AsyncProjectService.SampleRegistrationInformation;
+import life.qbic.projectmanagement.application.api.AsyncProjectService.SampleUpdateInformation;
+import org.apache.commons.collections.map.HashedMap;
+
+/**
+ * Converter Factory for creating {@link MetadataConverterV2} instances.
+ * <p>
+ * The converter factory is a registry for known converter implementation of the
+ * {@link MetadataConverterV2} interface.
+ * <p>
+ * It currently supports the following converters:
+ *
+ * <ul>
+ *   <li>{@link SampleRegistrationMetadataConverter}</li>
+ *   <li>{@link SampleUpdateMetadataConverter}</li>
+ *   <li>{@link MeasurementRegistrationMetadataConverterNGS}</li>
+ *   <li>{@link MeasurementRegistrationMetadataConverterPxP}</li>
+ *   <li>{@link MeasurementUpdateMetadataConverterNGS}</li>
+ *   <li>{@link MeasurementUpdateMetadataConverterPxP}</li>
+ * </ul>
+ *
+ * @since 1.10.0
+ */
+public class ConverterFactory {
```

Comment:
I think the name of the class should correspond to the pattern you used. The pattern being the Registry pattern instead of the factory pattern. You do not create a separate instance on the create method.
> Registry Design Pattern centralizes the creation and management of a global set of objects, providing a single point of access and ensuring controlled instantiation.

compared to

> [Factory](https://java-design-patterns.com/patterns/factory/): Used to encapsulate the instantiation logic that might be needed when objects are retrieved from the Registry.

https://java-design-patterns.com/patterns/registry/#programmatic-example-of-registry-pattern-in-java

Reply:
I do think using the registry pattern here is a great improvement :+1: